src/instructlab/data/generate.py

```diff
@@ -214,10 +222,25 @@ def generate(
         from instructlab.model.backends.llama_cpp import Server as llama_cpp_server
 
         # TODO (cdoern): we really should not edit the cfg object
-        ctx.obj.config.serve.llama_cpp.llm_family = model_family
-        backend_instance = backends.select_backend(
-            cfg=ctx.obj.config.generate.teacher, model_path=model_path
+        gen_cfg = copy.copy(ctx.obj.config)
+        gen_cfg.generate.teacher.llama_cpp.llm_family = model_family
+        if gpus is not None:
+            tps_prefix = "--tensor-parallel-size"
+            if backends.vllm.contains_argument(
+                tps_prefix, gen_cfg.generte.teacher.vllm.vllm_args
+            ):
+                click.secho(
+                    "Using gpus from --gpus or generate.teacher config and ignoring --tensor-parallel-size configured in generate.teacher vllm_args",
+                    fg="yellow",
+                )
+            gen_cfg.generate.teacher.vllm.vllm_args.extend([tps_prefix, str(gpus)])
```

Comment:
is this true? eval does not do this. I am pretty sure at long as there is a later version of the same flag in the list it used the LAST one right?

Reply:
The last instance of setting an argument will override right?  So I don't think you have to change the previous values.

Reply:
I just checked on vLLM directly and the last version of a repeated flag will be the one that applies. It still doesn't feel great to me that we could have duplicate arguments, in case vLLM somehow changes the way they handle that. But the logic is fine for now.